src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterFactory.cpp

```diff
@@ -456,6 +456,42 @@ DDSFilterFactory::~DDSFilterFactory()
     pool.clear();
 }
 
+static std::shared_ptr<xtypes::TypeObject> get_complete_type_object(
```

Comment:
I think this method would fit better into the `TypeObjectRegistry`.
In case the comment is accepted, we should doxygen the method

Reply:
Will do on a follow-up so we can backport this one